openlibrary/core/vendors.py

```diff
@@ -187,6 +206,16 @@ def clean_amazon_metadata_for_load(metadata):
     if metadata.get('source_records'):
         asin = metadata.get('source_records')[0].replace('amazon:', '')
         conforming_metadata['identifiers'] = {'amazon': [asin]}
+    title, subtitle = split_amazon_title(metadata['title'])
+    conforming_metadata['title'] == title
```

Comment:
```suggestion
    conforming_metadata['title'] = title
```

Reply:
oh wow, this could be the cause of the weird results I saw in some cases, great catch!